client/core/core.go

```diff
@@ -2497,3 +2497,25 @@ func validateOrderResponse(dc *dexConnection, result *msgjson.OrderResult, ord o
 	}
 	return nil
 }
+
+// Logout logs the user out
+func (c *Core) Logout() error {
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+	for _, asset := range c.user.Assets {
+		for _, dc := range c.conns {
+			if dc.hasOrders(asset.ID) {
+				return fmt.Errorf("cannot logs out with active orders, wallet %s ", unbip(asset.ID))
+			}
+			dc.acct.lock()
```

Comment:
This would lock the dex account before checking the other assets for orders.  Before you lock a dex account (unload the `dexAccount`'s private key), you want to check all of the assets against that dex connection.

Reply:
And just note that `range` over a map is a random order, so it testing this may be tricky.